### PR TITLE
Add Zeitwerk loader support

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

**Review by ekohl — APPROVED** [review]

File: lib/foreman_theme_satellite/engine.rb [hunk 1]
```diff
@@ -6,21 +6,11 @@
...
-
-    initializer 'foreman_theme_satellite.load_default_settings', :before => :load_config_initializers do |app|
-      SettingRegistry.prepend SettingRegistryBranding
-    end
+    config.eager_load_paths += Dir["#{config.root}/lib"]
 
     initializer 'foreman_theme_satellite.register_plugin', :before=> :finisher_hook do |app|
       Foreman::Plugin.register :foreman_theme_satellite do
-        requires_foreman '>= 3.7.0'
+        requires_foreman '>= 3.13.0'
```
Comment (evgeni) [inline lib/foreman_theme_satellite/engine.rb:13]: We run CI with 3.9, so I assume it will fail now.

Comment (ekohl) [reply lib/foreman_theme_satellite/engine.rb:13]: It certainly does.

**Comment by evgeni** [comment]
Any idea why the link test now fails?

**Comment by ofedoren** [comment]
Isn't this because something wasn't updated for the tests? Like https://github.com/RedHatSatellite/foreman_theme_satellite/pull/70

**Comment by evgeni** [comment]
No

**Comment by ekohl** [comment]
```
 NoMethodError: undefined method `empty?' for nil:NilClass
/home/runner/work/foreman_theme_satellite/foreman_theme_satellite/foreman_theme_satellite/test/link_checker.rb:50:in `navigate_path'
```

I think the way it gets loaded doesn't work. The rake task doesn't load the whole Rails environment but loads one file directly.

In https://github.com/RedHatSatellite/foreman_theme_satellite/pull/58 I've taken a more native approach, but I still have to finish that.

File: test/link_checker.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def navigate_path(path, hash = @toc)
 
     inner_hash = hash[first]
 
-    return inner_hash if rest.empty?
+    return inner_hash if rest.nil? or rest.empty?
```
Comment (evgeni) [inline test/link_checker.rb]: Before Zeitwerk, Rails was loaded when this test was running and Rails in its endless wisdom adds `empty?` to `NilClass`. With Zeitweirk, no Rails is present here, and so no `NilClass.empty?`. Let's catch that via `nil?` first.

**Review by ekohl — APPROVED** [review]

**Pull request merged**